### PR TITLE
Discussion: Proposals for new ways to interact with setUI

### PR DESCRIPTION
**Pull request by thyttan**
This is not meant to be merged - only for providing an example/proposal for changes to how we can interact with Bangle.setUI. This is an addition to the discussion at https://github.com/espruino/BangleApps/issues/3435.

The diff has comments explaining my thoughts.

### CODE REVIEW

**Comment by gfwilliams** [comment]
Thanks - extending to add the handlers looks good, but the whole `extra` thing seems like a bit of a mess. I think if you want to do something like that, it should really just be using `custom` and implementing its own behaviour

File: apps/spotrem/app.js [hunk 1]
```diff
@@ -98,25 +98,39 @@ let swipeHandler = function(LR, _) {
...
+       widgetUtils.show();
+      },
+      /**
+       * TODO: Proposal: Handlers added at the same level as the `mode : "updown"` entry
+       * will replace those that come with the mode - if there were any of that type.
+       * The handlers in the `extra` object below will still be added in addition to these ones.
+       **/
+      swipe : swipeHandler,
+      btn : buttonHandler,
+      btnEdge : "rising", // TODO: Proposal: a new, optional, way to choose what edge the button(-s) should act on.
```
Comment (thyttan) [inline apps/spotrem/app.js:118]: @gfwilliams what do you think of the idea to be able to specify the desired edge like this or some other way?

Comment (gfwilliams) [reply apps/spotrem/app.js:118]: Well, I was going to say this didn't feel great but I couldn't think of a better way...

But then I was thinking: "What if you want to do different things on Rising *and* falling edges?"

So what if we had `btn` (as before) but then `btnPress`/`btnRelease` callbacks could be specified? That feels like maybe a tidier solution? What do you think?

Comment (gfwilliams) [reply apps/spotrem/app.js:118]: ... the only downside of this is I guess it's not easily backwards compatible - but if we want this stuff to work outside of the `mode:custom` then maybe that doesn't matter as it wouldn't work now anyway

Comment (thyttan) [reply apps/spotrem/app.js:118]: I think I agree with your callbacks solution being tidier.

But just regarding this: 
> But then I was thinking: "What if you want to do different things on Rising and falling edges?"

I'm not saying it's the better solution, just another one. If "both" was specified in my example above that could be passed on to the `setWatch` call inside setUI and then the `buttonHandler` function would be responsible for handling both edges.

If you feel callbacks are better that carries much weight for me :)

Comment (thyttan) [reply apps/spotrem/app.js:118]: @gfwilliams 

I think I agree with your callbacks solution being tidier.

But just regarding this: 
> But then I was thinking: "What if you want to do different things on Rising and falling edges?"

I'm not saying it's the better solution, just another one. If "both" was specified in my example above that could be passed on to the `setWatch` call inside setUI and then the `buttonHandler` function would be responsible for handling both edges.

And if you feel callbacks are better that carries much weight for me :)

**Pull request closed without merging**